sdk/nodejs/runtime/invoke.ts

```diff
@@ -68,38 +68,7 @@ const providerproto = require("../proto/provider_pb.js");
  * synchronously.
  */
 export function invoke(tok: string, props: Inputs, opts: InvokeOptions = {}): Promise<any> {
-    if (opts.async) {
-        // User specifically requested async invoking.  Respect that.
-        return invokeAsync(tok, props, opts);
-    }
-
-    const config = new Config("pulumi");
-    const noSyncCalls = config.getBoolean("noSyncCalls");
-    if (noSyncCalls) {
-        // User globally disabled sync invokes.
-        return invokeAsync(tok, props, opts);
-    }
-
-    const syncResult = invokeSync(tok, props, opts);
-
-    // Wrap the synchronous value in a Promise view as well so that consumers can treat it
-    // either as the real value or something they can use as a Promise.
-    return createLiftedPromise(syncResult);
-}
-
-/**
- * Invokes the provided token *synchronously* no matter what.
- * @internal
- */
-export function invokeSync<T>(tok: string, props: Inputs, opts: InvokeOptions = {}): T {
-    const syncInvokes = tryGetSyncInvokes();
-    if (!syncInvokes) {
-        // We weren't launched from a pulumi CLI that supports sync-invokes.  Let the user know they
-        // should update and fall back to synchronously blocking on the async invoke.
-        return invokeFallbackToAsync<T>(tok, props, opts);
-    }
-
-    return invokeSyncWorker<T>(tok, props, opts, syncInvokes);
+    return invokeAsync(tok, props, opts);
```

Comment:
Do we strictly have to couple these two changes together?  Even if one is dependent on the other - it might make sense to land independently.  

If the grpc change really is dependent on removing support for sync invokes - then we likely can't land it till the 2.0 branch.  Is there any way to decouple these at all?

Reply:
I'd really, really like to see these changes decoupled as well.

Reply:
i'll see what i can do here.

Reply:
I removed the deasync part.

Reply:
#resolved